### PR TITLE
Adding fix for billing header and test cases

### PR DESCRIPTION
**Pull request by joeysabs**
## What Does this PR Do?
Fix for billing header bug where different subscription intervals are only appearing as monthly. Additionally, some test cases to go along with the file

---
## Summary by cubic
Fixes the billing header showing all subscriptions as monthly. Interval labels now derive from purchase data with a price fallback and correctly render “daily.”

- **Bug Fixes**
  - intervalLabel now falls back to price.intervalCount/unit when purchase is missing.
  - Fixed label generation: pluralizes for counts > 1 and uses “daily” (not “dayly”).
  - pricingSubtitleForSubscriptionFlow now passes price into intervalLabel.
  - Added unit tests for daily/weekly/monthly/yearly, multi-intervals, fallbacks, and zero-count edge cases.

### CODE REVIEW

File: platform/flowglad-next/src/components/checkout/billing-header.test.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+vi.mock('@/utils/stripe', () => ({
+  stripeCurrencyAmountToHumanReadableCurrencyAmount: vi.fn((currency, amount) => {
+    return `$${(amount / 100).toFixed(2)}`
+  }),
+}))
+
+const mockUseCheckoutPageContext = vi.mocked(useCheckoutPageContext)
+
+describe('BillingHeader', () => {
+  const mockCheckoutContext: Partial<CheckoutPageContextValues> = {
```
Comment (BrooksFlannery) [inline platform/flowglad-next/src/components/checkout/billing-header.test.tsx]: this is unused

**Review by BrooksFlannery — APPROVED** [review]
